packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,269 @@
+<Avatar square icon={<IDBadgeIcon />} />
+```
+
+Displaying a badge (\*\*subject to change pending final spec for the `Badge` component):
+
+```jsx
+<Avatar name="Allan Munger" badge={<PresenceBadge status="busy" />} />
+```
+
+With active state indication:
+
+```jsx
+<Avatar name="Daisy Phillips" active={true} activeDisplay="ring-shadow" />
+<Avatar name="Robin Counts" active={false} activeDisplay="ring-shadow" />
+```
+
+## Variants
+
+### Shape
+
+The Avatar natively supports a circular and square shape. However, some use cases require a hexagon shape (e.g. to represent a bot). The hexagon will not be supported directly, but it is possible to style the Avatar with a hexagon background by using `tokens`:
+
+```jsx
+<Avatar
+  icon={<ChatBotIcon />}
+  display="icon"
+  tokens={{
+    width: 'calc(var(--avatar-height) * 1.125)',
+    background: `url('${hexagonSvg}') 0px/contain no-repeat`,
+    borderRadius: '0',
+  }}
+/>
+```
+
+Where hexagonSvg is defined as:
+
+```js
+hexagonSvg =
+  'data:image/svg+xml;utf8,' +
+  '<svg width="36" height="32" viewBox="0 0 36 32" fill="none" xmlns="http://www.w3.org/2000/svg">' +
+  '<path fill="rgb(232,232,232)" d="M0.407926 17.528C-0.135976 16.5859 -0.135975 15.4141 0.407926 14.472' +
+  'L7.91541 1.46793C8.44076 0.557947 9.39444 0 10.4245 0H25.5755C26.6056 0 27.5592 0.557951 28.0846 1.46793' +
+  'L35.5921 14.472C36.136 15.4141 36.136 16.5859 35.5921 17.528L28.0846 30.5321' +
+  'C27.5592 31.4421 26.6056 32 25.5755 32H10.4245C9.39443 32 8.44076 31.4421 7.91541 30.5321L0.407926 17.528Z"/>' +
+  '</svg>';
+```
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /** What the avatar displays. This can be used to override the default behavior, for example to show the icon even
+   *  if the avatar has initials that could be shown as the label. */
+  display?: 'image' | 'label' | 'icon';
+
+  /**
+   * Optional activity indicator
+   * * true: the avatar will be decorated according to activeDisplay
+   * * false: the avatar will be reduced in size and partially transparent
+   * * undefined (default): normal display
+   */
+  active?: boolean;
```

Comment:
Wanted to get feedback about the `active` prop. It is a tri-state, with different visuals for true, false, or undefined.

Are we ok with having a tri-state prop as a `boolean?`, or should it be an enum like:
`active?: 'normal' | 'active' | 'inactive';`
(not sure of a good name for `'normal'`; or it could be left off, and the normal visuals apply whenever the prop is unset, like it is today)

Reply:
I think an enum would be more explicit.

As for the name, I think it could explicitly be `unset`.

Reply:
Do you mean the literal string `'unset'`? As in `active="unset"`? (Though that would be the default so in practice nobody would need to write that)